src/handlers/install_handler.rs

```diff
@@ -642,6 +617,44 @@ async fn handle_building_from_source(version: &ParsedVersion, config: &Config) -
     Ok(PostDownloadVersionType::Hash)
 }
 
+#[rustfmt::skip]
+async fn remote_checks(non_parsed_str: &str) -> Result<()> {
+    // check if repo has a remote
+    let remote = Command::new("git").arg("remote").arg("get-url").arg("origin").stdout(Stdio::null()).spawn()?.wait().await?;
+    if !remote.success() {
+        // add neovim's remote
+        Command::new("git").arg("remote").arg("add").arg("origin").arg("https://github.com/neovim/neovim.git").spawn()?.wait().await?;
+    } else {
+        // set neovim's remote otherwise
+        Command::new("git").arg("remote").arg("set-url").arg("origin").arg("https://github.com/neovim/neovim.git").spawn()?.wait().await?;
+    }
+
+    // fetch version from origin
+    let fetch_successful = Command::new("git").arg("fetch").arg("--depth").arg("1").arg("origin").arg(non_parsed_str).spawn()?.wait().await?.success();
+
+    if !fetch_successful {
+        return Err(anyhow!(
+            "fetching remote failed, try providing the full commit hash"
+        ));
+    }
+
+    // checkout fetched files
+    Command::new("git").arg("checkout").arg("FETCH_HEAD").stdout(Stdio::null()).spawn()?.wait().await?;
+    Ok(())
+}
+
+async fn metadata_handler<F, Fut>(dir: &str, action: F) -> Result<()>
```

Comment:
This could be potentially removed and changed to Path::exists, i opted to use exists checks via metadata cuz sometimes the exists function didnt work correctly.